parlai/agents/rag/retrievers.py

```diff
@@ -1305,10 +1305,17 @@ def __init__(self, opt: Opt, dictionary: DictionaryAgent, shared: TShared = None
         self.n_docs = opt['n_docs']
         self._query_ids = dict()
         self._saved_docs = dict()
+        self._largest_seen_idx = -1
         super().__init__(opt, dictionary, shared=shared)
 
     def add_retrieve_doc(self, query: str, retrieved_docs: List[Document]):
-        new_idx = len(self._query_ids)
+        self._largest_seen_idx += 1
+        new_idx = max(len(self._query_ids), self._largest_seen_idx)
```

Comment:
If we are using `self._largest_seen_idx` as a counter that gets reset afterward, why do we still need `len(self._query_ids)`?

Reply:
ah ok u r right, we don't. fixxing that